failed to build with boost

Package: influxdb-cxx[boost,core]:arm64-osx@0.7.3

**Host Environment**

- Host: arm64-osx
- Compiler: AppleClang 16.0.0.16000026
-    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: b918f110a2 2025-03-16 (6 minutes ago)

**To Reproduce**

`vcpkg install --enforce-port-checks --allow-unsupported`

**Failure logs**

```
-- Using cached offa-influxdb-cxx-v0.7.3.tar.gz
-- Extracting source /Users/leanderSchulten/git_projekte/vcpkg2/downloads/offa-influxdb-cxx-v0.7.3.tar.gz
-- Applying patch fix-dllexports.patch
-- Using source at /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean
-- Found external ninja('1.12.1').
-- Configuring arm64-osx
-- Building arm64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /Users/leanderSchulten/git_projekte/vcpkg2/downloads/tools/cmake-3.30.1-osx/cmake-3.30.1-macos-universal/CMake.app/Contents/bin/cmake --build . --config Debug --target install -- -v -j11
    Working Directory: /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg
    See logs for more information:
      /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/install-arm64-osx-dbg-out.log

Call Stack (most recent call first):
  vcpkg_installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  vcpkg_installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/influxdb-cxx/portfile.cmake:24 (vcpkg_cmake_install)
  scripts/ports.cmake:203 (include)



```

<details><summary>/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/install-arm64-osx-dbg-out.log</summary>

```
Change Dir: '/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg'

Run Build Command(s): /opt/homebrew/bin/ninja -v -v -j11 install
[1/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCURL_STATICLIB -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-Core.dir/Proxy.cxx.o -MF src/CMakeFiles/InfluxDB-Core.dir/Proxy.cxx.o.d -o src/CMakeFiles/InfluxDB-Core.dir/Proxy.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/Proxy.cxx
[2/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCURL_STATICLIB -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-Internal.dir/LineProtocol.cxx.o -MF src/CMakeFiles/InfluxDB-Internal.dir/LineProtocol.cxx.o.d -o src/CMakeFiles/InfluxDB-Internal.dir/LineProtocol.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/LineProtocol.cxx
[3/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCURL_STATICLIB -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-Core.dir/InfluxDB.cxx.o -MF src/CMakeFiles/InfluxDB-Core.dir/InfluxDB.cxx.o.d -o src/CMakeFiles/InfluxDB-Core.dir/InfluxDB.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/InfluxDB.cxx
[4/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCURL_STATICLIB -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-Core.dir/Point.cxx.o -MF src/CMakeFiles/InfluxDB-Core.dir/Point.cxx.o.d -o src/CMakeFiles/InfluxDB-Core.dir/Point.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/Point.cxx
[5/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCURL_STATICLIB -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-Internal.dir/HTTP.cxx.o -MF src/CMakeFiles/InfluxDB-Internal.dir/HTTP.cxx.o.d -o src/CMakeFiles/InfluxDB-Internal.dir/HTTP.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/HTTP.cxx
[6/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCURL_STATICLIB -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-Core.dir/InfluxDBFactory.cxx.o -MF src/CMakeFiles/InfluxDB-Core.dir/InfluxDBFactory.cxx.o.d -o src/CMakeFiles/InfluxDB-Core.dir/InfluxDBFactory.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/InfluxDBFactory.cxx
[7/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/3rd-party -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-BoostSupport.dir/UnixSocket.cxx.o -MF src/CMakeFiles/InfluxDB-BoostSupport.dir/UnixSocket.cxx.o.d -o src/CMakeFiles/InfluxDB-BoostSupport.dir/UnixSocket.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UnixSocket.cxx
FAILED: src/CMakeFiles/InfluxDB-BoostSupport.dir/UnixSocket.cxx.o 
ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/3rd-party -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-BoostSupport.dir/UnixSocket.cxx.o -MF src/CMakeFiles/InfluxDB-BoostSupport.dir/UnixSocket.cxx.o.d -o src/CMakeFiles/InfluxDB-BoostSupport.dir/UnixSocket.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UnixSocket.cxx
In file included from /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UnixSocket.cxx:28:
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UnixSocket.h:50:22: error: no type named 'io_service' in namespace 'boost::asio'
   50 |         boost::asio::io_service mIoService;
      |         ~~~~~~~~~~~~~^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UnixSocket.h:50:33: error: private field 'mIoService' is not used [-Werror,-Wunused-private-field]
   50 |         boost::asio::io_service mIoService;
      |                                 ^
2 errors generated.
[8/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/3rd-party -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-BoostSupport.dir/UDP.cxx.o -MF src/CMakeFiles/InfluxDB-BoostSupport.dir/UDP.cxx.o.d -o src/CMakeFiles/InfluxDB-BoostSupport.dir/UDP.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx
FAILED: src/CMakeFiles/InfluxDB-BoostSupport.dir/UDP.cxx.o 
ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/3rd-party -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-BoostSupport.dir/UDP.cxx.o -MF src/CMakeFiles/InfluxDB-BoostSupport.dir/UDP.cxx.o.d -o src/CMakeFiles/InfluxDB-BoostSupport.dir/UDP.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx
In file included from /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx:28:
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.h:52:22: error: no type named 'io_service' in namespace 'boost::asio'
   52 |         boost::asio::io_service mIoService;
      |         ~~~~~~~~~~~~~^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx:39:46: error: expected ';' after expression
   39 |         boost::asio::ip::udp::resolver::query query(boost::asio::ip::udp::v4(), hostname, std::to_string(port));
      |                                              ^
      |                                              ;
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx:39:9: error: no member named 'query' in 'boost::asio::ip::basic_resolver<boost::asio::ip::udp>'; did you mean simply 'query'?
   39 |         boost::asio::ip::udp::resolver::query query(boost::asio::ip::udp::v4(), hostname, std::to_string(port));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         query
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include/Transport.h:50:29: note: 'query' declared here
   50 |         virtual std::string query([[maybe_unused]] const std::string& query)
      |                             ^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx:39:47: error: cannot initialize object parameter of type 'influxdb::Transport' with an expression of type 'influxdb::transports::UDP'
   39 |         boost::asio::ip::udp::resolver::query query(boost::asio::ip::udp::v4(), hostname, std::to_string(port));
      |                                               ^~~~~
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx:40:41: error: no type named 'iterator' in 'boost::asio::ip::basic_resolver<boost::asio::ip::udp>'
   40 |         boost::asio::ip::udp::resolver::iterator resolverInerator = resolver.resolve(query);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx:40:86: error: reference to non-static member function must be called
   40 |         boost::asio::ip::udp::resolver::iterator resolverInerator = resolver.resolve(query);
      |                                                                                      ^~~~~
In file included from /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.cxx:28:
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.h:52:33: error: private field 'mIoService' is not used [-Werror,-Wunused-private-field]
   52 |         boost::asio::io_service mIoService;
      |                                 ^
7 errors generated.
[9/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/3rd-party -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-BoostSupport.dir/TCP.cxx.o -MF src/CMakeFiles/InfluxDB-BoostSupport.dir/TCP.cxx.o.d -o src/CMakeFiles/InfluxDB-BoostSupport.dir/TCP.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx
FAILED: src/CMakeFiles/InfluxDB-BoostSupport.dir/TCP.cxx.o 
ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/3rd-party -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-BoostSupport.dir/TCP.cxx.o -MF src/CMakeFiles/InfluxDB-BoostSupport.dir/TCP.cxx.o.d -o src/CMakeFiles/InfluxDB-BoostSupport.dir/TCP.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx
In file included from /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:27:
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.h:57:22: error: no type named 'io_service' in namespace 'boost::asio'
   57 |         boost::asio::io_service mIoService;
      |         ~~~~~~~~~~~~~^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:39:37: error: expected ';' after expression
   39 |         ba::ip::tcp::resolver::query query(hostname, std::to_string(port));
      |                                     ^
      |                                     ;
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:39:9: error: no member named 'query' in 'boost::asio::ip::basic_resolver<boost::asio::ip::tcp>'; did you mean simply 'query'?
   39 |         ba::ip::tcp::resolver::query query(hostname, std::to_string(port));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         query
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include/Transport.h:50:29: note: 'query' declared here
   50 |         virtual std::string query([[maybe_unused]] const std::string& query)
      |                             ^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:39:38: error: cannot initialize object parameter of type 'influxdb::Transport' with an expression of type 'influxdb::transports::TCP'
   39 |         ba::ip::tcp::resolver::query query(hostname, std::to_string(port));
      |                                      ^~~~~
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:40:32: error: no type named 'iterator' in 'boost::asio::ip::basic_resolver<boost::asio::ip::tcp>'
   40 |         ba::ip::tcp::resolver::iterator resolverIterator = resolver.resolve(query);
      |         ~~~~~~~~~~~~~~~~~~~~~~~^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:40:77: error: reference to non-static member function must be called
   40 |         ba::ip::tcp::resolver::iterator resolverIterator = resolver.resolve(query);
      |                                                                             ^~~~~
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:41:41: error: use of undeclared identifier 'end'; did you mean 'std::end'?
   41 |         ba::ip::tcp::resolver::iterator end;
      |                                         ^~~
      |                                         std::end
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__iterator/access.h:28:46: note: 'std::end' declared here
   28 | _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR _Tp* end(_Tp (&__array)[_Np]) _NOEXCEPT {
      |                                              ^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:41:32: error: no member named 'iterator' in 'boost::asio::ip::basic_resolver<boost::asio::ip::tcp>'; did you mean 'operator'?
   41 |         ba::ip::tcp::resolver::iterator end;
      |         ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
      |                                operator
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:41:41: error: expected a type
   41 |         ba::ip::tcp::resolver::iterator end;
      |                                         ^
In file included from /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.cxx:27:
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.h:57:33: error: private field 'mIoService' is not used [-Werror,-Wunused-private-field]
   57 |         boost::asio::io_service mIoService;
      |                                 ^
10 errors generated.
[10/12] ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/3rd-party -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-BoostSupport.dir/BoostSupport.cxx.o -MF src/CMakeFiles/InfluxDB-BoostSupport.dir/BoostSupport.cxx.o.d -o src/CMakeFiles/InfluxDB-BoostSupport.dir/BoostSupport.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/BoostSupport.cxx
FAILED: src/CMakeFiles/InfluxDB-BoostSupport.dir/BoostSupport.cxx.o 
ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/include -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/arm64-osx-dbg/src -I/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/3rd-party -isystem /Users/leanderSchulten/git_projekte/vcpkg2/vcpkg_installed/arm64-osx/include -fPIC -g -std=c++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -pedantic -pedantic-errors -Werror -Wshadow -Wold-style-cast -Wnull-dereference -Wnon-virtual-dtor -Woverloaded-virtual -MD -MT src/CMakeFiles/InfluxDB-BoostSupport.dir/BoostSupport.cxx.o -MF src/CMakeFiles/InfluxDB-BoostSupport.dir/BoostSupport.cxx.o.d -o src/CMakeFiles/InfluxDB-BoostSupport.dir/BoostSupport.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/BoostSupport.cxx
In file included from /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/BoostSupport.cxx:25:
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UDP.h:52:22: error: no type named 'io_service' in namespace 'boost::asio'
   52 |         boost::asio::io_service mIoService;
      |         ~~~~~~~~~~~~~^
In file included from /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/BoostSupport.cxx:26:
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/TCP.h:57:22: error: no type named 'io_service' in namespace 'boost::asio'
   57 |         boost::asio::io_service mIoService;
      |         ~~~~~~~~~~~~~^
In file included from /Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/BoostSupport.cxx:27:
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/UnixSocket.h:50:22: error: no type named 'io_service' in namespace 'boost::asio'
   50 |         boost::asio::io_service mIoService;
      |         ~~~~~~~~~~~~~^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/BoostSupport.cxx:106:16: error: no viable conversion from returned value of type 'unique_ptr<influxdb::transports::UDP>' to function return type 'unique_ptr<Transport>'
  106 |         return std::make_unique<transports::UDP>(uri.host, uri.port);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:124:59: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'typename __unique_if<UDP>::__unique_single' (aka 'unique_ptr<influxdb::transports::UDP>') to 'const unique_ptr<Transport> &' for 1st argument
  124 | class _LIBCPP_UNIQUE_PTR_TRIVIAL_ABI _LIBCPP_TEMPLATE_VIS unique_ptr {
      |                                                           ^~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:170:43: note: candidate constructor template not viable: no known conversion from 'typename __unique_if<UDP>::__unique_single' (aka 'unique_ptr<influxdb::transports::UDP>') to 'nullptr_t' (aka 'std::nullptr_t') for 1st argument
  170 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR unique_ptr(nullptr_t) _NOEXCEPT
      |                                           ^          ~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:190:55: note: candidate constructor not viable: no known conversion from 'typename __unique_if<UDP>::__unique_single' (aka 'unique_ptr<influxdb::transports::UDP>') to 'unique_ptr<Transport> &&' for 1st argument
  190 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 unique_ptr(unique_ptr&& __u) _NOEXCEPT
      |                                                       ^          ~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:197:55: note: candidate template ignored: requirement 'is_convertible<influxdb::transports::UDP *, influxdb::Transport *>::value' was not satisfied [with _Up = influxdb::transports::UDP, _Ep = std::default_delete<influxdb::transports::UDP>]
  197 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 unique_ptr(unique_ptr<_Up, _Ep>&& __u) _NOEXCEPT
      |                                                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:174:64: note: explicit constructor is not a candidate
  174 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 explicit unique_ptr(pointer __p) _NOEXCEPT
      |                                                                ^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/BoostSupport.cxx:111:16: error: no viable conversion from returned value of type 'unique_ptr<influxdb::transports::TCP>' to function return type 'unique_ptr<Transport>'
  111 |         return std::make_unique<transports::TCP>(uri.host, uri.port);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:124:59: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'typename __unique_if<TCP>::__unique_single' (aka 'unique_ptr<influxdb::transports::TCP>') to 'const unique_ptr<Transport> &' for 1st argument
  124 | class _LIBCPP_UNIQUE_PTR_TRIVIAL_ABI _LIBCPP_TEMPLATE_VIS unique_ptr {
      |                                                           ^~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:170:43: note: candidate constructor template not viable: no known conversion from 'typename __unique_if<TCP>::__unique_single' (aka 'unique_ptr<influxdb::transports::TCP>') to 'nullptr_t' (aka 'std::nullptr_t') for 1st argument
  170 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR unique_ptr(nullptr_t) _NOEXCEPT
      |                                           ^          ~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:190:55: note: candidate constructor not viable: no known conversion from 'typename __unique_if<TCP>::__unique_single' (aka 'unique_ptr<influxdb::transports::TCP>') to 'unique_ptr<Transport> &&' for 1st argument
  190 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 unique_ptr(unique_ptr&& __u) _NOEXCEPT
      |                                                       ^          ~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:197:55: note: candidate template ignored: requirement 'is_convertible<influxdb::transports::TCP *, influxdb::Transport *>::value' was not satisfied [with _Up = influxdb::transports::TCP, _Ep = std::default_delete<influxdb::transports::TCP>]
  197 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 unique_ptr(unique_ptr<_Up, _Ep>&& __u) _NOEXCEPT
      |                                                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:174:64: note: explicit constructor is not a candidate
  174 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 explicit unique_ptr(pointer __p) _NOEXCEPT
      |                                                                ^
/Users/leanderSchulten/git_projekte/vcpkg2/buildtrees/influxdb-cxx/src/v0.7.3-0482e18de6.clean/src/BoostSupport.cxx:116:16: error: no viable conversion from returned value of type 'unique_ptr<influxdb::transports::UnixSocket>' to function return type 'unique_ptr<Transport>'
  116 |         return std::make_unique<transports::UnixSocket>(uri.path);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:124:59: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'typename __unique_if<UnixSocket>::__unique_single' (aka 'unique_ptr<influxdb::transports::UnixSocket>') to 'const unique_ptr<Transport> &' for 1st argument
  124 | class _LIBCPP_UNIQUE_PTR_TRIVIAL_ABI _LIBCPP_TEMPLATE_VIS unique_ptr {
      |                                                           ^~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:170:43: note: candidate constructor template not viable: no known conversion from 'typename __unique_if<UnixSocket>::__unique_single' (aka 'unique_ptr<influxdb::transports::UnixSocket>') to 'nullptr_t' (aka 'std::nullptr_t') for 1st argument
  170 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR unique_ptr(nullptr_t) _NOEXCEPT
      |                                           ^          ~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:190:55: note: candidate constructor not viable: no known conversion from 'typename __unique_if<UnixSocket>::__unique_single' (aka 'unique_ptr<influxdb::transports::UnixSocket>') to 'unique_ptr<Transport> &&' for 1st argument
  190 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 unique_ptr(unique_ptr&& __u) _NOEXCEPT
      |                                                       ^          ~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:197:55: note: candidate template ignored: requirement 'is_convertible<influxdb::transports::UnixSocket *, influxdb::Transport *>::value' was not satisfied [with _Up = influxdb::transports::UnixSocket, _Ep = std::default_delete<influxdb::transports::UnixSocket>]
  197 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 unique_ptr(unique_ptr<_Up, _Ep>&& __u) _NOEXCEPT
      |                                                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/usr/include/c++/v1/__memory/unique_ptr.h:174:64: note: explicit constructor is not a candidate
  174 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 explicit unique_ptr(pointer __p) _NOEXCEPT
      |                                                                ^
6 errors generated.
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "dependencies": [
    {
      "name": "influxdb-cxx",
      "default-features": false,
      "features": [
        "boost"
      ]
    }
  ],
  "$builtin-baseline": "edc84ff66e6262a9f7565c28eb76577aeab9c5aa",
  "$overrides": [
    {
      "name": "cgal",
      "version": "5.6.1"
    }
  ]
}

```
</details>

--- a/src/TCP.cxx
+++ b/src/TCP.cxx
@@ -30,16 +30,16 @@
 
 namespace influxdb::transports
 {
-    namespace ba = boost::asio;
-
     TCP::TCP(const std::string& hostname, int port)
-        : mSocket(mIoService)
+        : mSocket(mIoContext)
     {
-        ba::ip::tcp::resolver resolver(mIoService);
-        ba::ip::tcp::resolver::query query(hostname, std::to_string(port));
-        ba::ip::tcp::resolver::iterator resolverIterator = resolver.resolve(query);
-        ba::ip::tcp::resolver::iterator end;
-        mEndpoint = *resolverIterator;
+        boost::asio::ip::tcp::resolver resolver(mIoContext);
+        mEndpoint = *(resolver
+                          .resolve(boost::asio::ip::tcp::v4(),
+                                   hostname,
+                                   std::to_string(port),
+                                   boost::asio::ip::resolver_query_base::passive)
+                          .cbegin());
         mSocket.open(mEndpoint.protocol());
         reconnect();
     }
@@ -52,7 +52,7 @@ namespace influxdb::transports
     void TCP::reconnect()
     {
         mSocket.connect(mEndpoint);
-        mSocket.wait(ba::ip::tcp::socket::wait_write);
+        mSocket.wait(boost::asio::ip::tcp::socket::wait_write);
     }
 
     void TCP::send(std::string&& message)
@@ -60,7 +60,7 @@ namespace influxdb::transports
         try
         {
             message.append("\n");
-            const size_t written = mSocket.write_some(ba::buffer(message, message.size()));
+            const size_t written = mSocket.write_some(boost::asio::buffer(message, message.size()));
             if (written != message.size())
             {
                 throw InfluxDBException("Error while transmitting data");

--- a/src/TCP.h
+++ b/src/TCP.h
@@ -54,7 +54,7 @@ namespace influxdb::transports
 
     private:
         /// Boost Asio I/O functionality
-        boost::asio::io_service mIoService;
+        boost::asio::io_context mIoContext;
 
         /// TCP socket
         boost::asio::ip::tcp::socket mSocket;

--- a/src/UDP.cxx
+++ b/src/UDP.cxx
@@ -33,12 +33,15 @@ namespace influxdb::transports
 {
 
     UDP::UDP(const std::string& hostname, int port)
-        : mSocket(mIoService, boost::asio::ip::udp::endpoint(boost::asio::ip::udp::v4(), 0))
+        : mSocket(mIoContext, boost::asio::ip::udp::endpoint(boost::asio::ip::udp::v4(), 0))
     {
-        boost::asio::ip::udp::resolver resolver(mIoService);
-        boost::asio::ip::udp::resolver::query query(boost::asio::ip::udp::v4(), hostname, std::to_string(port));
-        boost::asio::ip::udp::resolver::iterator resolverInerator = resolver.resolve(query);
-        mEndpoint = *resolverInerator;
+        boost::asio::ip::udp::resolver resolver(mIoContext);
+        mEndpoint = *(resolver
+                          .resolve(boost::asio::ip::udp::v4(),
+                                   hostname,
+                                   std::to_string(port),
+                                   boost::asio::ip::resolver_query_base::passive)
+                          .cbegin());
     }
 
     void UDP::send(std::string&& message)

--- a/src/UDP.h
+++ b/src/UDP.h
@@ -49,7 +49,7 @@ namespace influxdb::transports
 
     private:
         /// Boost Asio I/O functionality
-        boost::asio::io_service mIoService;
+        boost::asio::io_context mIoContext;
 
         /// UDP socket
         boost::asio::ip::udp::socket mSocket;

--- a/src/UnixSocket.cxx
+++ b/src/UnixSocket.cxx
@@ -34,7 +34,7 @@ namespace influxdb::transports
 #if defined(BOOST_ASIO_HAS_LOCAL_SOCKETS)
 
     UnixSocket::UnixSocket(const std::string& socketPath)
-        : mSocket(mIoService), mEndpoint(socketPath)
+        : mSocket(mIoContext), mEndpoint(socketPath)
     {
         mSocket.open();
     }

--- a/src/UnixSocket.h
+++ b/src/UnixSocket.h
@@ -47,7 +47,7 @@ namespace influxdb::transports
 
     private:
         /// Boost Asio I/O functionality
-        boost::asio::io_service mIoService;
+        boost::asio::io_context mIoContext;
 #if defined(BOOST_ASIO_HAS_LOCAL_SOCKETS)
         /// Unix socket
         boost::asio::local::datagram_protocol::socket mSocket;

